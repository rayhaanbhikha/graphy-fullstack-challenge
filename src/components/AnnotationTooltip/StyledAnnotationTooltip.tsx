import styled from "styled-components";

export const StyledAnnotationTooltip = styled.div`
  background: rgb(216, 216, 216);
  min-width: 200px;
  min-height: 100px;
  border-radius: 3px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  border: solid 1px rgb(173,173,173);
`