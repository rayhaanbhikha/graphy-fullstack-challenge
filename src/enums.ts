
export enum ApplicationState {
  EDIT_MODE,
  DEFAULT_MODE
}

export enum AnnotationStates {
  OPEN,
  CLOSED,
  EDITING,
  DELETING,
  DRAGGING
}