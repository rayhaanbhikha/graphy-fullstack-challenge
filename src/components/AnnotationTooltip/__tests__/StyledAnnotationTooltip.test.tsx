import React from 'react';
import { render } from '@testing-library/react'
import { StyledAnnotationTooltip } from '../StyledAnnotationTooltip';
import 'jest-styled-components';
import { AnnotationStates } from '../../../enums';

describe('StyledAnnotationTooltip', () => {

  it('should have correct default styles', () => {
    const { asFragment } = render(<StyledAnnotationTooltip annotationState={AnnotationStates.CLOSED} />);
    expect(asFragment()).toMatchSnapshot();
  })

  it('should have correct styles when in edit state', () => {
    const { asFragment } = render(<StyledAnnotationTooltip annotationState={AnnotationStates.EDITING} />);
    expect(asFragment()).toMatchSnapshot();
  })

  it('should have correct styles when in dragging state', () => {
    const { asFragment } = render(<StyledAnnotationTooltip annotationState={AnnotationStates.DRAGGING} />);
    expect(asFragment()).toMatchSnapshot();
  })
});