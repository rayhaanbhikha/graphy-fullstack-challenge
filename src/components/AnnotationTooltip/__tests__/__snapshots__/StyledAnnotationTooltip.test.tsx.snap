// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledAnnotationTooltip should have correct default styles 1`] = `
<DocumentFragment>
  .c0 {
  background: rgb(216,216,216);
  min-width: 200px;
  min-height: 100px;
  border-radius: 3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border: solid 1px rgb(173,173,173);
  padding: 3px;
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`StyledAnnotationTooltip should have correct styles when in dragging state 1`] = `
<DocumentFragment>
  .c0 {
  background: rgb(216,216,216);
  min-width: 200px;
  min-height: 100px;
  border-radius: 3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border: solid 1px rgb(173,173,173);
  padding: 3px;
  cursor: -webkit-grabbing;
  cursor: -moz-grabbing;
  cursor: grabbing;
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`StyledAnnotationTooltip should have correct styles when in edit state 1`] = `
<DocumentFragment>
  .c0 {
  background: rgb(216,216,216);
  min-width: 200px;
  min-height: 100px;
  border-radius: 3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border: solid 1px rgb(173,173,173);
  padding: 3px;
  cursor: initial;
}

<div
    class="c0"
  />
</DocumentFragment>
`;
